File: 06_Inheritance/Basics.java
// Base Class
class Animal {
    void eat() {
        System.out.println("This animal eats food.");
    }
}

// Derived Class
class Dog extends Animal {
    void bark() {
        System.out.println("The dog barks.");
    }
}

public class Basics {
    public static void main(String[] args) {
        Dog dog = new Dog();
        dog.eat();  // Inherited method
        dog.bark(); // Own method
    }
}

File: 06_Inheritance/Basics.cpp
#include <iostream>
using namespace std;

class Animal {
protected:
    void eat() {
        cout << "This animal eats food." << endl;
    }
};

class Dog : public Animal {
public:
    void bark() {
        cout << "The dog barks." << endl;
    }

    void eat() {
        Animal::eat();
    }
};

int main() {
    Animal animal;
    Dog dog;

    dog.bark(); 
    dog.eat();
    return 0;
}
